testscript/testscript.go

```diff
@@ -788,6 +789,27 @@ func (ts *TestScript) Check(err error) {
 	}
 }
 
+// Stdout returns an io.Writer that can be used to update the last stdout.
+// This is typically used by commands declared via Params.Cmds.
+func (ts *TestScript) Stdout() io.Writer {
+	return stringAppender{&ts.stdout}
+}
+
+// Stdout returns an io.Writer that can be used to update the last stderr.
+// This is typically used by commands declared via Params.Cmds.
+func (ts *TestScript) Stderr() io.Writer {
+	return stringAppender{&ts.stderr}
+}
+
+type stringAppender struct {
+	s *string
+}
+
+func (sa stringAppender) Write(p []byte) (n int, err error) {
+	*sa.s = *sa.s + string(p)
```

Comment:
perhaps leave a TODO to use a strings.Builder at some point; this is going to be quadratic and pretty inefficient.

Reply:
I'm not clear that we can use `strings.Builder`. The underlying `stdout` field is a `string`, and there is a lot of usage of that field assuming it is a `string`. i.e. it would be a significant change in the semantics of the underlying field if we were to make that change.

Perhaps we should, therefore, expose `(*TestScript).SetStdout(string)` instead. Because it better reflects what's going on.

Reply:
Alternative approach in https://github.com/rogpeppe/go-internal/pull/217. Indeed I think I prefer that approach because it places the burden of "gathering" stdout on the caller.

Reply:
I slightly prefer io.Writer APIs, but I agree it goes against the internal "string value" design, so using the other PR seems fine to me. Another disadvantage of this API is that `TestScript.Stdout` could be understood as a stdout getter/reader rather than setter or writer, and `TestScript.SetStdout` doesn't have that issue.